### PR TITLE
Fix(Onesignal): Code refactor

### PR DESCRIPTION
**Pull request by Deokumarjnu**
# Description

Added setTimeout in one signal script to prevent error in lighthouse scores (Unused javascript error, also reducing TTI) etc.

Fixes # https://github.com/quintype/ace-planning/issues/197

Dependencies # NA

Documentation # (link to the corresponding documentation changes)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Released beta version and tested it on malibu-advanced prod

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by ags1773 — APPROVED** [review]

File: server/views/partial/one-signal.ejs [hunk 1]
```diff
@@ -1,5 +1,11 @@
-<script src="https://cdn.onesignal.com/sdks/OneSignalSDK.js" async></script>
 <script>
+  (() => setTimeout(() => {
```
Comment (sharangj) [inline server/views/partial/one-signal.ejs:2]: Instead of this can't we add it in startApp()

File: server/views/partial/one-signal.ejs [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+    const myScript = document.createElement("script");
+    myScript.setAttribute("src", "https://cdn.onesignal.com/sdks/OneSignalSDK.js");
```
Comment (sharangj) [inline server/views/partial/one-signal.ejs:4]: What happens if oneSignal is not used?

File: server/views/partial/one-signal.ejs [hunk 3]
```diff
@@ -1,5 +1,11 @@
-<script src="https://cdn.onesignal.com/sdks/OneSignalSDK.js" async></script>
 <script>
+  (() => setTimeout(() => {
+    const myScript = document.createElement("script");
```
Comment (sharangj) [inline server/views/partial/one-signal.ejs:3]: Please name properly.

**Comment by sharangj** [comment]
I am super glad that we are thinking of making things configurable in the framework. Kudos @nandap4790 @Jeevan-Kishore @Deokumarjnu.

File: server/views/partial/one-signal.ejs (hunk 3 above)
Comment (nandap4790) [reply server/views/partial/one-signal.ejs:3]: done

File: server/views/partial/one-signal.ejs (hunk 2 above)
Comment (nandap4790) [reply server/views/partial/one-signal.ejs:4]: https://github.com/quintype/malibu-advanced/blob/onesignal-refactor/app/server/handlers/render-layout.js#L75
https://github.com/quintype/malibu-advanced/blob/onesignal-refactor/views/pages/layout.ejs#L43

File: server/views/partial/one-signal.ejs (hunk 1 above)
Comment (sharangj) [reply server/views/partial/one-signal.ejs:2]: Cant we move this to startApp()?

**Pull request merged**